subprojects/core/src/main/java/org/gradle/api/internal/AbstractTask.java

```diff
@@ -121,6 +121,8 @@
 
     private final DefaultTaskDependency dependencies;
 
+    private final DefaultTaskDependency explicitDependsOns;
```

Comment:
This reminds me of a related plan to phase out `dependsOn()` for tasks in general, and promote adding dependencies via task inputs only. We'd like to introduce a special `LifecycleTask` type for things like `assemble` and `check` that would keep their `dependsOn()` method, and would in turn be forbidden to declare actions or inputs/outputs. I should write this up someday in a spec.

Anyway, the reason I'm mentioning this is because it is related, and maybe we should call this field `lifecycleDependencies`, WDYT?

Reply:
I like the name, it feels descriptive and future-proof.